src/Moq/Mock.cs

```diff
@@ -566,14 +566,11 @@ private static VoidSetupPhrase<T> SetupSetPexProtected<T>(Mock<T> mock, Action<T
 			return new SetterSetupPhrase<T, TProperty>(setup);
 		}
 
-		private static (Mock, LambdaExpression, MethodInfo, Expression[]) SetupSetImpl<T>(
-			Mock<T> mock,
-			Action<T> setterExpression)
-			where T : class
+		private static (Mock, LambdaExpression, MethodInfo, Expression[]) SetupSetImpl(Mock mock, Delegate setterExpression)
 		{
 			using (var context = new FluentMockContext())
 			{
-				setterExpression(mock.Object);
+				setterExpression.DynamicInvoke(mock.Object);
```

Comment:
There would be ways around `DynamicInvoke` but they are probably not worth the effort. (For example, we could use an `Action<object>` instead of a `Delegate`, then convert the parameter between `T` and `object` in the calling methods. However, this would also require capturing the original delegate's parameter name and passing it into `SetupSetImpl` as well; c.f. the lines of code below this one.)